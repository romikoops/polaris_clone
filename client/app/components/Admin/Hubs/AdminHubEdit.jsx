import React, { Component } from 'react'
// import Select from 'react-select'
// import styled from 'styled-components'
import PropTypes from '../../../prop-types'
import styles from './index.scss'
import GmapsWrapper from '../../../hocs/GmapsWrapper'
import PlaceSearch from '../../Maps/PlaceSearch'
import '../../../styles/select-css-custom.css'
import SquareButton from '../../SquareButton'
import { gradientTextGenerator } from '../../../helpers'

export class AdminHubEdit extends Component {
  constructor (props) {
    super(props)
    this.state = {
      location: props.hub.location,
      hub: props.hub
    }
    this.handlePlaceChange = this.handlePlaceChange.bind(this)
    this.handleAddressChange = this.handleAddressChange.bind(this)
    this.resetAuto = this.resetAuto.bind(this)
    this.saveHubEdit = this.saveHubEdit.bind(this)
    this.handleTruckingType = this.handleTruckingType.bind(this)
    this.handleHubType = this.handleHubType.bind(this)
  }
  componentDidMount () {
    if (this.props.hub) {
      this.setExistingLocation(this.props.hub.location)
    }
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.hub && !this.state.hub.id) {
      this.setState({ hub: nextProps.hub })
    }
    if (nextProps.hub && nextProps.hub.location && !this.state.location.id) {
      this.setExistingLocation(nextProps.hub.location)
    }
  }
  setExistingLocation (locationObj) {
    const location = {
      ...locationObj,
      country: locationObj.country.name
    }
    this.setState({ location })
  }
  handlePlaceChange (place) {
    const tmpAddress = {
      number: '',
      street: '',
      zipCode: '',
      city: '',
      country: '',
      fullAddress: ''
    }

    place.address_components.forEach((ac) => {
      if (ac.types.includes('street_number')) {
        tmpAddress.street_number = ac.long_name
      }

      if (ac.types.includes('route') || ac.types.includes('premise')) {
        tmpAddress.street = ac.long_name
      }

      if (ac.types.includes('locality')) {
        tmpAddress.city = ac.long_name
      }

      if (ac.types.includes('postal_code')) {
        tmpAddress.zipCode = ac.long_name
      }

      if (ac.types.includes('country')) {
        tmpAddress.country = ac.long_name
      }
    })
    tmpAddress.latitude = place.geometry.location.lat()
    tmpAddress.longitude = place.geometry.location.lng()
    tmpAddress.fullAddress = place.formatted_address
    tmpAddress.geocoded_address = place.formatted_address
    this.setState({
      location: tmpAddress,
      hub: {
        ...this.state.hub,
        name: tmpAddress.city
      },
      autocomplete: { ...this.state.autocomplete, location: true }
    })
  }
  handleAddressChange (event) {
    const eventKeys = event.target.name.split('-')
    const key1 = eventKeys[0]
    const key2 = eventKeys[1]
    const val = event.target.value
    this.setState({
      [key1]: {
        ...this.state[key1],
        [key2]: val
      }
    })
  }
  handleImageUpload (e) {
    const { adminDispatch, hub } = this.props
    const file = e.target.files[0]
    adminDispatch.newHubImage(hub.id, file)
  }

  clickUploaderInput (e) {
    e.preventDefault()
    this.uploaderInput.click()
  }
  resetAuto () {
    this.setState({
      autocomplete: { ...this.state.autocomplete, location: false }
    })
  }
  handleTruckingType (ev) {
    this.setState({
      hub: {
        ...this.state.hub,
        truckingType: ev.value
      }
    })
  }
  handleHubType (ev) {
    const { hub } = this.state
    let newName
    if (hub.name) {
      newName = `${hub.name} ${ev.label}`
    } else {
      newName = ''
    }

    this.setState({
      hub: {
        ...this.state.hub,
        name: newName,
        hubType: ev.value
      }
    })
  }

  saveHubEdit () {
    const { hub, location } = this.state
    const preppedLocation = {}
    const preppedHub = {}
    preppedLocation.street_number = location.street_number
    preppedLocation.street = location.street
    preppedLocation.zip_code = location.zip_code
    preppedLocation.city = location.city
    preppedLocation.country = location.country
    preppedLocation.latitude = location.latitude
    preppedLocation.longitude = location.longitude
    preppedLocation.geocoded_address = location.geocoded_address

    preppedHub.name = hub.name
    preppedHub.latitude = location.latitude
    preppedHub.longitude = location.longitude
    this.props.adminDispatch
      .editHub(hub.id, { data: preppedHub, location: preppedLocation })
    this.props.close()
  }

  render () {
    const { theme, close } = this.props
    const { hub, location } = this.state
    const iconStyle = gradientTextGenerator(theme.colors.primary, theme.colors.secondary)

    return (
      <div
        className={`flex-none layout-align-center-center layout-row ${styles.editor_backdrop}`}
      >
        <div className={`flex-none ${styles.editor_fade}`} onClick={() => close()} />
        <div
          className={`${styles.hub_form} layout-row flex-none layout-wrap layout-align-center`}
        >
          <div className="flex-100 layout-row layout-wrap layout-align-start-center">
            <div className="flex-5" />
            <h2 className="flex-none letter_3 no_m" >
              Edit Hub
            </h2>
          </div>
          <div className="flex-100 layout-row layout-align-start-center layout-wrap">
            <GmapsWrapper
              theme={theme}
              component={PlaceSearch}
              handlePlaceChange={this.handlePlaceChange}
              location={hub.location}
            />
          </div>
          <div
            className={`flex-100 layout-row layout-wrap layout-align-start-center ${
              styles.form_padding
            }`}
          >
            <div className="flex-50 layout-row layout-wrap layout-align-start-start height_100">
              <div className="flex-100 layout-row layout-wrap layout-align-start-start">

                <div
                  className={`flex-80 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}
                >
                  <p className="flex-100">Hub Name</p>
                  <input
                    name="hub-name"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={hub.name}
                    placeholder="Name"
                  />
                </div>
              </div>
              <div className="flex-100 layout-row layout-align-start-start">
                <div className="flex-50 layout-row layout-wrap layout-align-center-center">

                  <div
                    className={`flex-80 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}
                  >
                    <p className="flex-100">Latitude</p>
                    <input
                      name="location-latitude"
                      className="flex-none"
                      type="string"
                      onChange={this.handleAddressChange}
                      value={location.latitude}
                      placeholder="Latitude"
                    />
                  </div>
                </div>
                <div className="flex-50 layout-row layout-wrap layout-align-center-center">

                  <div
                    className={`flex-80 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}
                  >
                    <p className="flex-100">Longitude</p>
                    <input
                      name="location-longitude"
                      className="flex-none"
                      type="string"
                      onChange={this.handleAddressChange}
                      value={location.longitude}
                      placeholder="Longitude"
                    />
                  </div>
                </div>

              </div>
              <div className="flex-100 layout-row layout-wrap layout-align-center-start">
                <div className={`flex-none layout-row ${styles.upload_btn_wrapper} `}>
                  <form>
                    <div
                      className={`${styles.upload_image} flex-none pointy layout-row layout-align-space-around-center`}
                      onClick={e => this.clickUploaderInput(e)}
                    >
                      <p className={`${styles.upload_title}`}>Upload New Image</p>
                      <i className="fa fa-cloud-upload clip flex-none" style={iconStyle} />
                    </div>
                    <input
                      type="file"
                      onChange={e => this.handleImageUpload(e)}
                      name="hub_image"
                      ref={(input) => {
                        this.uploaderInput = input
                      }}
                    />
                  </form>
                </div>
              </div>

            </div>
            <div className="flex-50 layout-row layout-wrap layout-align-end-space-around">
              <div className="flex-100 layout-row layout-align-space-around-center">
                <div className={`flex-20 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}>
                  <p className="flex-100">No.</p>
                  <input
                    id="not-auto"
                    name="location-street_number"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={location.street_number}
                    placeholder="Number"
                  />
                </div>
                <div className={`flex-75 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}>
                  <p className="flex-100">Street</p>
                  <input
                    name="location-street"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={location.street}
                    placeholder="Street"
                  />
                </div>
              </div>
              <div className="flex-100 layout-row layout-align-space-around-center">
                <div className={`flex-30 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}>
                  <p className="flex-100">Zipcode</p>
                  <input
                    name="location-zipCode"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={location.zip_code}
                    placeholder="Zip Code"
                  />
                </div>
                <div className={`flex-65 layout-row layout-wrap ayout-align-center-center ${styles.material_input}`}>
                  <p className="flex-100">City</p>
                  <input
                    name="location-city"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={location.city}
                    placeholder="City"
                  />
                </div>
              </div>

              <div className="flex-100 layout-row layout-align-space-around-center">
                <div className={`flex-100 layout-row layout-wrap layout-align-center-center ${styles.material_input}`}>
                  <p className="flex-100">Country</p>
                  <input
                    name="location-country"
                    className="flex-none"
                    type="string"
                    onChange={this.handleAddressChange}
                    value={location.country}
                    placeholder="Country"
                  />
                </div>
              </div>
            </div>
          </div>
          <div className="flex-100 layout-row layout-wrap layout-align-center-center">
            <div className="flex-none layout-row">
              <SquareButton
                theme={theme}
                size="small"
                text="Save Hub"

                handleNext={this.saveHubEdit}
                iconClass="fa-floppy"
              />
            </div>
            <div className="flex-5" />
          </div>
        </div>
      </div>
    )
  }
}

AdminHubEdit.propTypes = {
  theme: PropTypes.theme,
  hub: PropTypes.hub.isRequired,
  close: PropTypes.func.isRequired,
  adminDispatch: PropTypes.objectOf(PropTypes.func)
}

AdminHubEdit.defaultProps = {
  theme: null,
  adminDispatch: {}
}

export default AdminHubEdit
