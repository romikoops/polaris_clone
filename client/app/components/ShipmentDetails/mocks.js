// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
