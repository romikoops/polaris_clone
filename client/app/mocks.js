import { set, cloneDeep } from 'lodash'

/**
 * Checks if variable is an non-empty object
 *
 * True for `{a:1}`
 * Fase for `null, {}, [], [1,2,3]`
 */
const isObject = (x) => {
  const ok = x !== null && !Array.isArray(x) && typeof x === 'object'
  if (!ok) {
    return false
  }

  return Object.keys(x).length > 0
}
/* eslint-disable */
/**
 * Used in unit test to modify specific properties
 * with minimal lines of code
 *
 * You can specify `path` to be empty string
 * if the change affects more than one of object's branches
 */
export const change = (origin, pathRaw, rules) => {
  const willReturn = cloneDeep(origin)
  
  if (!isObject(rules)) {
    set(willReturn, pathRaw, rules)
    return willReturn
  }
  const path = pathRaw === '' ? '' : `${pathRaw}.`
  for (const ruleKey of Object.keys(rules)) {
    const rule = rules[ruleKey]
    if (!isObject(rule)) {
      set(
        willReturn,
        `${path}${ruleKey}`,
        rule
      )
      continue
    }
    Object.keys(rule).filter(subruleKey => !isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      set(
        willReturn,
        `${path}${ruleKey}.${subruleKey}`,
        subrule
      )
    })
    Object.keys(rule).filter(subruleKey => isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      Object.keys(subrule).map((deepKey) => {
        const deep = rule[subruleKey][deepKey]
        set(
          willReturn,
          `${path}${ruleKey}.${subruleKey}.${deepKey}`,
          deep
        )
      })
    })
  }

  return willReturn
}
/* eslint-enable */
export const identity = input => input

export const theme = {
  colors: {
    primary: '#333',
    secondary: '#fafafa'
  }
}

export const user = {
  role: { name: 'shipper' },
  company_name: 'FOO_COMPANY',
  currency: 'EUR',
  email: 'foo@bar.baz',
  first_name: 'John',
  guest: false,
  last_name: 'Doe',
  phone: '6345789'
}

export const history = {
  push: identity
}

export const tenant = {
  data: {
    id: 123,
    scope: {
      modes_of_transport: {
        ocean: {
          OCEAN_LOAD_TYPE: true
        },
        air: {},
        truck: {},
        rail: {}
      }
    },
    theme,
    subdomain: 'foosubdomain'
  }
}

export const req = {
  schedule: [{}],
  total: 345,
  shipment: {}
}

const schedulesInShipmentData = [
  { hub_route_key: 'OCEAN_HUB_ROUTE_KEY', mode_of_transport: 'ocean' },
  { hub_route_key: 'AIR_HUB_ROUTE_KEY', mode_of_transport: 'air' },
  { hub_route_key: 'TRUCK_HUB_ROUTE_KEY', mode_of_transport: 'truck' },
  { hub_route_key: 'RAIL_HUB_ROUTE_KEY', mode_of_transport: 'rail' }
]

export const shipmentInShipmentData = {
  load_type: 'OCEAN_LOAD_TYPE',
  selected_offer: { cargo: {}, total: { value: 77 } },
  schedules_charges: {
    OCEAN_HUB_ROUTE_KEY: { total: 40 },
    AIR_HUB_ROUTE_KEY: { total: 200 },
    TRUCK_HUB_ROUTE_KEY: { total: 75 },
    RAIL_HUB_ROUTE_KEY: { total: 125 }
  },
  total_goods_value: {
    value: 15,
    currency: 'USD'
  },
  trucking: {
    pre_carriage: { trucking_time_in_seconds: 55 }
  },
  has_on_carriage: false,
  has_pre_carriage: false,
  notes: 'FOO_NOTES',
  cargo_notes: 'FOO_CARGO_NOTES',
  eori: 1234,
  incoterm_text: 'FOO_INCOTERM_TEXT',
  total_price: {
    value: 12,
    currency: 'USD'
  }
}

export const shipmentData = {
  contacts: [],
  shipment: shipmentInShipmentData,
  documents: [],
  cargoItems: [],
  containers: [],
  schedules: schedulesInShipmentData,
  locations: {
    startHub: 'FOO_START_HUB',
    endHub: 'FOO_END_HUB',
    destination: {
      street_number: 9,
      street: 'BAR_STREET',
      city: 'BAR_CITY',
      country: 'China',
      zip_code: 845321
    },
    origin: {
      street_number: 7,
      street: 'FOO_STREET',
      city: 'FOO_CITY',
      country: 'Germany',
      zip_code: 21177
    }
  }
}

export const match = {
  params: {}
}

export const shipments = {
  direction: 'export',
  finished: [],
  open: [],
  requested: []
}

export const shipment = {
  origin_hub: { name: 'FOO_ORIGIN_HUB' },
  destination_hub: { name: 'FOO_DESTINATION_HUB' },
  id: 654,
  status: 'FOO_STATUS',
  clientName: 'FOO_CLIENT_NAME',
  planned_etd: 789,
  imc_reference: 'FOO_IMC_REFERENCE',
  schedule_set: []
}

export const location = {
  primary: true,
  id: 246,
  street_number: 579,
  streetNumber: 579,
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_STREET',
  city: 'FOO_CITY',
  zip_code: '22456',
  zipCode: '22456',
  country: 'Germany'
}

export const address = {
  id: 13579,
  street_number: '1',
  street: 'Uhlandweg',
  city: 'Hamburg',
  zip_code: '22848',
  country: 'Germany'
}

export const hub = {
  location,
  name: 'FOO_HUB_NAME'
}

export const client = {
  first_name: 'Sebastian',
  last_name: 'Muller',
  email: 'muller55@yahoo.de',
  company_name: 'Mehristweniger',
  password: 'pwd'
}

export const route = {
  id: 24680
}

export const vehicleType = {
  id: 827364,
  name: 'FOO_VEHICLE_TYPE_NAME',
  mode_of_transport: 'FOO_VEHICLE_TYPE_TRANSPORT'
}

export const schedule = {
  hub_route_key: 'FOO_SCHEDULE_ROUTE_KEY',
  id: 555777,
  eta: 441188
}

export const charge = {
  id: 654297
}

export const locations = {
  origin: {},
  destination: {}
}

class MapMock {
  constructor (x) {
    this.x = x
  }
  bindTo () {
    return this.x
  }
  setContent () {
    return this.x
  }
  addListener () {
    return this.x
  }
}

export const gMaps = {
  InfoWindow: MapMock,
  LatLngBounds: MapMock,
  Map: MapMock,
  MapTypeId: { ROADMAP: '' },
  Marker: MapMock,
  Point: MapMock,
  Size: MapMock,
  places: { Autocomplete: MapMock }
}
