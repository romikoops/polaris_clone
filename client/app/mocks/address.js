export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
