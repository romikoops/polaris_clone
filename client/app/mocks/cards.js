export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
