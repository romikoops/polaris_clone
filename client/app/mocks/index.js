export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
