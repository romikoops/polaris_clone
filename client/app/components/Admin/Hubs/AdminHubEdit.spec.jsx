import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity, hub } from '../../../mocks'

import AdminHubEdit from './AdminHubEdit'

const propsBase = {
  hub,
  close: identity,
  adminDispatch: {},
  theme
}

test('shallow render', () => {
  expect(shallow(<AdminHubEdit {...propsBase} />)).toMatchSnapshot()
})
