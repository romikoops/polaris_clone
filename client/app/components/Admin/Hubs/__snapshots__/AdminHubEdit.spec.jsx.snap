// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="flex-none layout-align-center-center layout-row editor_backdrop"
>
  <div
    className="flex-none editor_fade"
    onClick={[Function]}
  />
  <div
    className="hub_form layout-row flex-none layout-wrap layout-align-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-start-center"
    >
      <div
        className="flex-5"
      />
      <h2
        className="flex-none letter_3 no_m"
      >
        Edit Hub
      </h2>
    </div>
    <div
      className="flex-100 layout-row layout-align-start-center layout-wrap"
    >
      <GmapsWrapper
        address={
          Object {
            "city": "FOO_CITY",
            "country": "Germany",
            "geocodedAddress": "FOO_GEOCODED_ADDRESS",
            "id": 246,
            "primary": true,
            "street": "FOO_STREET",
            "streetNumber": 579,
            "street_number": 579,
            "zipCode": "22456",
            "zip_code": "22456",
          }
        }
        component={[Function]}
        handlePlaceChange={[Function]}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-start-center form_padding"
    >
      <div
        className="flex-50 layout-row layout-wrap layout-align-start-start height_100"
      >
        <div
          className="flex-100 layout-row layout-wrap layout-align-start-start"
        >
          <div
            className="flex-80 layout-row layout-wrap layout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              Hub Name
            </p>
            <input
              className="flex-none"
              name="hub-name"
              onChange={[Function]}
              placeholder="Name"
              type="string"
              value="FOO_HUB_NAME"
            />
          </div>
        </div>
        <div
          className="flex-100 layout-row layout-align-start-start"
        >
          <div
            className="flex-50 layout-row layout-wrap layout-align-center-center"
          >
            <div
              className="flex-80 layout-row layout-wrap layout-align-center-center material_input"
            >
              <p
                className="flex-100"
              >
                Latitude
              </p>
              <input
                className="flex-none"
                name="address-latitude"
                onChange={[Function]}
                placeholder="Latitude"
                type="string"
              />
            </div>
          </div>
          <div
            className="flex-50 layout-row layout-wrap layout-align-center-center"
          >
            <div
              className="flex-80 layout-row layout-wrap layout-align-center-center material_input"
            >
              <p
                className="flex-100"
              >
                Longitude
              </p>
              <input
                className="flex-none"
                name="address-longitude"
                onChange={[Function]}
                placeholder="Longitude"
                type="string"
              />
            </div>
          </div>
        </div>
        <div
          className="flex-100 layout-row layout-wrap layout-align-center-start"
        >
          <div
            className="flex-none layout-row upload_btn_wrapper "
          >
            <form>
              <div
                className="upload_image flex-none pointy layout-row layout-align-space-around-center"
                onClick={[Function]}
              >
                <p
                  className="upload_title"
                >
                  Upload New Image
                </p>
                <i
                  className="fa fa-cloud-upload clip flex-none"
                  style={
                    Object {
                      "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
                    }
                  }
                />
              </div>
              <input
                name="hub_image"
                onChange={[Function]}
                type="file"
              />
            </form>
          </div>
        </div>
      </div>
      <div
        className="flex-50 layout-row layout-wrap layout-align-end-space-around"
      >
        <div
          className="flex-100 layout-row layout-align-space-around-center"
        >
          <div
            className="flex-20 layout-row layout-wrap layout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              No.
            </p>
            <input
              className="flex-none"
              id="not-auto"
              name="address-street_number"
              onChange={[Function]}
              placeholder="Number"
              type="string"
              value={579}
            />
          </div>
          <div
            className="flex-75 layout-row layout-wrap layout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              Street
            </p>
            <input
              className="flex-none"
              name="address-street"
              onChange={[Function]}
              placeholder="Street"
              type="string"
              value="FOO_STREET"
            />
          </div>
        </div>
        <div
          className="flex-100 layout-row layout-align-space-around-center"
        >
          <div
            className="flex-30 layout-row layout-wrap layout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              Zipcode
            </p>
            <input
              className="flex-none"
              name="address-zipCode"
              onChange={[Function]}
              placeholder="Zip Code"
              type="string"
              value="22456"
            />
          </div>
          <div
            className="flex-65 layout-row layout-wrap ayout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              City
            </p>
            <input
              className="flex-none"
              name="address-city"
              onChange={[Function]}
              placeholder="City"
              type="string"
              value="FOO_CITY"
            />
          </div>
        </div>
        <div
          className="flex-100 layout-row layout-align-space-around-center"
        >
          <div
            className="flex-100 layout-row layout-wrap layout-align-center-center material_input"
          >
            <p
              className="flex-100"
            >
              Country
            </p>
            <input
              className="flex-none"
              name="address-country"
              onChange={[Function]}
              placeholder="Country"
              type="string"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-center"
    >
      <div
        className="flex-none layout-row"
      >
        <SquareButton
          active={false}
          back={false}
          border={false}
          classNames=""
          disabled={false}
          handleDisabled={null}
          handleNext={[Function]}
          icon=""
          iconClass="fa-floppy"
          size="small"
          text="Save Hub"
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
        />
      </div>
      <div
        className="flex-5"
      />
    </div>
  </div>
</div>
`;
